internal/pkg/otel/run.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+const buildDescription = "Elastic opentelemetry-collector distribution"
+
+// IsOtelConfig returns true if file is named:
+//   - otel.(yaml|yml)
+//   - otlp.(yaml|yml)
+//   - otelcol.(yaml|yml)
+//
+// in case of other filename it checks for `agent`, `inputs`, `outputs` keyword. If at least one of these is present
+// it assumes agent config and returns (false, nil).
+// If these keywords are missing it checks for presence of otel specific keywords `exporters`, `receivers`,`processors`
+// or `extensions` in combination with `service` definition. In case these are found (true, nil) is returned.
+// If none of these conditions are satisfied (false, nil) is returned assuming agent config for backward compatibility.
+func IsOtelConfig(ctx context.Context, pathConfigFile string) (bool, error) {
+	fileName := filepath.Base(pathConfigFile)
+	cleanFileName := strings.TrimSpace(strings.ToLower(strings.TrimSuffix(fileName, filepath.Ext(fileName))))
+	if cleanFileName == "otel" || cleanFileName == "otlp" || cleanFileName == "otelcol" {
+		return true, nil
+	}
+
+	if suffix := filepath.Ext(fileName); suffix != ".yml" && suffix != ".yaml" {
+		return false, nil
+	}
+
+	rawConfig, err := config.LoadFile(pathConfigFile)
+	if err != nil {
+		return false, errors.New(err,
+			fmt.Sprintf("could not read configuration file %s", pathConfigFile),
+			errors.TypeFilesystem,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	mapConfig, err := rawConfig.ToMapStr()
+	if err != nil {
+		return false, errors.New(err,
+			errors.TypeConfig,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	// contains agent definition
+	_, hasAgent := mapConfig["agent"]
+	_, hasInputs := mapConfig["inputs"]
+	_, hasOutputs := mapConfig["outputs"]
+
+	if hasAgent || hasInputs || hasOutputs {
+		return false, nil
+	}
+
+	// contains otel service definition
+	_, hasService := mapConfig["service"]
+	_, hasExporters := mapConfig["exporters"]
+	_, hasReceivers := mapConfig["receivers"]
+	_, hasProcessors := mapConfig["processors"]
+	_, hasExtensions := mapConfig["extensions"]
+
+	if hasService && (hasExporters || hasReceivers || hasProcessors || hasExtensions) {
+		return true, nil
+	}
+
+	// default behavior is Elastic Agent
+	return false, nil
+}
+
+func Run(ctx context.Context, cancel context.CancelFunc, stop chan bool, testingMode bool) error {
+	fmt.Fprintln(os.Stdout, "Starting in otel mode")
+	settings, err := newSettings(paths.ConfigFile(), release.Version())
+	if err != nil {
+		return err
+	}
+
+	svc, err := otelcol.NewCollector(*settings)
+	if err != nil {
+		return err
+	}
+
+	// cancel context on stop from event manager
+	cancelCtx, cancel := context.WithCancel(ctx)
+	go func() {
+		<-stop
+		cancel()
+	}()
+	defer cancel()
+
+	return svc.Run(cancelCtx)
+}
+
+func newSettings(configPaths string, version string) (*otelcol.CollectorSettings, error) {
```

Comment:
Nit: rename to `configPath` (singular) since we're only expecting one path here now.